File: 41-FirstMissingPositive/41-FirstMissingPositive.java
// Last updated: 5/29/2026, 12:00:38 PM
class Solution {
    public int firstMissingPositive(int[] nums) {
        int len=nums.length;
        HashSet<Integer> hs= new HashSet<>();
        for(int i=0;i<nums.length;i++){
            hs.add(nums[i]);
        }
        for(int i=1;i<=nums.length;i++){
            if(!hs.contains(i)){
                return i;
            }
        }
        return len+1;
    }
}

File: 41-FirstMissingPositive/41-FirstMissingPositive.py
# Last updated: 5/29/2026, 12:00:32 PM
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        for i in range(len(nums)):
                while 1<=nums[i]<=len(nums) and nums[nums[i]-1]!=nums[i]:
                    temp=nums[nums[i]-1]
                    nums[nums[i]-1]=nums[i]
                    nums[i]=temp
        for i in range(len(nums)):
            if i+1!=nums[i]:
                return i+1
        return len(nums)+1